staff_features/accessions/step_definitions/accession_edit.rb

```diff
@@ -0,0 +1,188 @@
+
+    @accession_id = URI.decode_www_form_component(element[:href]).split('/').pop
+  end
+end
+
+Given 'the Accession is opened in the view mode' do
+  visit "#{STAFF_URL}/accessions"
+
+  fill_in 'filter-text', with: @uuid
+
+  within '.search-filter' do
+    find('button').click
+  end
+
+  search_result_rows = all('#tabledSearchResults tbody tr')
+  expect(search_result_rows.length).to eq 1
+
+  within search_result_rows[0] do
+    element = find('a', text: 'View')
+
+    @accession_id = URI.decode_www_form_component(element[:href]).split('/').pop
+  end
+
+  click_on 'View'
+end
+
+Given 'the Accession is opened in edit mode' do
+  visit "#{STAFF_URL}/accessions"
+
+  fill_in 'filter-text', with: @uuid
+
+  within '.search-filter' do
+    find('button').click
+  end
+
+  search_result_rows = all('#tabledSearchResults tbody tr')
+  expect(search_result_rows.length).to eq 1
+
+  within search_result_rows[0] do
+    element = find('a', text: 'Edit')
+
+    @accession_id = URI.decode_www_form_component(element[:href]).split('/').pop
+  end
+
+  click_on 'Edit'
+end
+
+When 'the user changes the {string} field to {string}' do |field, value|
+  fill_in field, with: value
+end
+
+When 'the user changes the {string} field' do |field|
+  fill_in field, with: "Updated Title #{SecureRandom.uuid}"
+end
+
+Then 'the {string} field has the original value' do |field|
+  expect(page).to have_field(field, with: "Accession Title #{@uuid}")
+end
+
+Then 'the Accession is updated with the new {string} as {string}' do |field, value|
+  fill_in field, with: value
+end
+
+Then 'the Accession is opened in the edit mode' do
+  uri = current_url.split('/')
+  uri.pop
+  accession_id = uri.pop
+
+  expect(accession_id).to eq @accession_id
+end
+
+Then 'all fields become editable' do
+  fill_in 'Identifier', with: 'Identifier'
+end
+
+Then 'the field {string} has value {string}' do |field, value|
+  expect(page).to have_field(field, with: value)
+end
+
+Then 'the Accession Date field has the original value' do
+  visit "#{STAFF_URL}/accessions/#{@accession_id}/edit"
+
+  expect(page).to have_field('Accession Date', with: '2000-01-01')
+end
+
+When 'the Identifier field has the original value' do
+  visit "#{STAFF_URL}/accessions/#{@accession_id}/edit"
+
+  expect(page).to have_field('Identifier', with: "Accession #{@uuid}")
+end
+
+Given 'the Accession is opened in edit mode by User A' do
+  visit "#{STAFF_URL}/accessions"
+
+  fill_in 'filter-text', with: @uuid
+
+  within '.search-filter' do
+    find('button').click
+  end
+
+  search_result_rows = all('#tabledSearchResults tbody tr')
+  expect(search_result_rows.length).to eq 1
+
+  within search_result_rows[0] do
+    element = find('a', text: 'Edit')
+
+    @accession_id = URI.decode_www_form_component(element[:href]).split('/').pop
+  end
+
+  click_on 'Edit'
+end
+
+Given 'the Accession is opened in edit mode by User B' do
+  @session = Capybara::Session.new(:firefox_alternative_session)
```

Comment:
this is maybe nitpicking, but I think it would be clearer to name this variable something that reflects that it is an alternative or headless session

Reply:
Yes this makes sense. I wouldn't say it's nitpicking. It's good to have clarity and self-explanatory variable names in the code. This alternative session is defined [here](https://github.com/archivesspace/e2e-tests/blob/9a66f0eef7c0171e4c62466193e54e5a0d8671c1/env.rb#L32-L37), and it may be headless or not, according to a command line option.

I have renamed this to `@user_b_session`, since all steps that make use of this session, refer to `User B`. Please let me know if you have a different name in mind.

Reply:
Yes this makes sense. I wouldn't say it's nitpicking. It's good to have clarity and self-explanatory variable names in the code. This alternative session is defined [here](https://github.com/archivesspace/e2e-tests/blob/9a66f0eef7c0171e4c62466193e54e5a0d8671c1/env.rb#L32-L37), and it may be headless or not, according to a command line option.

I have renamed this to `@user_b_session`, since all steps that make use of this session, refer to `User B`. Please let me know if you have a different name in mind.